srtcore/queue.cpp

```diff
@@ -943,107 +940,108 @@ void CRendezvousQueue::updateConnStatus(EReadStatus rst, EConnectStatus cst, con
-                 */
-                if (AF_INET == i->m_iIPversion)
-                    delete (sockaddr_in*)i->m_pPeerAddr;
-                else
-                    delete (sockaddr_in6*)i->m_pPeerAddr;
-
-                // i_next was preincremented, but this is guaranteed to point to
-                // the element next to erased one.
-                i_next = m_lRendezvousID.erase(i);
-                continue;
+                read_st = RST_AGAIN;
+                conn_st = CONN_AGAIN;
             }
 
-            // This queue is used only in case of Async mode (rendezvous or caller-listener).
-            // Synchronous connection requests are handled in startConnect() completely.
-            if (!i->m_pUDT->m_bSynRecving)
+            if (!i->m_pUDT->processAsyncConnectRequest(read_st, conn_st, response, i->m_pPeerAddr))
             {
+                // cst == CONN_REJECT can only be result of worker_ProcessAddressedPacket and
+                // its already set in this case.
+                LOGC(mglog.Error, log << "RendezvousQueue: processAsyncConnectRequest FAILED. Setting TTL as EXPIRED.");
+                i->m_pUDT->sendCtrl(UMSG_SHUTDOWN);
+                i->m_ullTTL = 0; // Make it expire right now, will be picked up at the next iteration
 #if ENABLE_HEAVY_LOGGING
-                ++debug_nupd;
+                ++debug_nfail;
 #endif
-                // IMPORTANT INFORMATION concerning changes towards UDT legacy.
-                // In the UDT code there was no attempt to interpret any incoming data.
-                // All data from the incoming packet were considered to be already deployed into
-                // m_ConnRes field, and m_ConnReq field was considered at this time accordingly updated.
-                // Therefore this procedure did only one thing: craft a new handshake packet and send it.
-                // In SRT this may also interpret extra data (extensions in case when Agent is Responder)
-                // and the `response` packet may sometimes contain no data. Therefore the passed `rst`
-                // must be checked to distinguish the call by periodic update (RST_AGAIN) from a call
-                // due to have received the packet (RST_OK).
-                //
-                // In the below call, only the underlying `processRendezvous` function will be attempting
-                // to interpret these data (for caller-listener this was already done by `processConnectRequest`
-                // before calling this function), and it checks for the data presence.
-                bool success;
-                if (i->m_iID != response.m_iID)
-                { 
-                    success = i->m_pUDT->processAsyncConnectRequest(RST_AGAIN, CONN_AGAIN, response, i->m_pPeerAddr);
-                }
-                else
-                {
-                    success = i->m_pUDT->processAsyncConnectRequest(rst, cst, response, i->m_pPeerAddr);
```

Comment:
Here two calls to `processAsyncConnectRequest` differ by two arguments.
Thus reduced to one call (line 1023):
`if (!i->m_pUDT->processAsyncConnectRequest(read_st, conn_st, response, i->m_pPeerAddr))`
`read_st` and `conn_st` are changed when required.

Reply:
Not sure if `cst` and `rst` variables aren't in use some odd way. Otherwise should work.